![Tinder](https://full-stack-assets.s3.eu-west-3.amazonaws.com/M03-EDA/Tinder-Symbole.png)

# Speed Dating with Tinder

## Goals 🎯

Use the dataset to understand what makes people interested into each other to go on a second date together:
* You may use descriptive statistics
* You may use visualisations

# PART 1: EXTRACTION AND DESCRIPTION

In [29]:
# 0. IMPORTER LES LIBRAIRIES

import pandas as pd
import numpy as np
import plotly.express as px
import plotly.graph_objects as go
from plotly.subplots import make_subplots

In [30]:
# 1. CHARGEMENT ET DESCRIPTION DU DATASET
# ----------------------------------------------------------------

df = pd.read_csv("Speed+Dating+Data.csv", encoding="latin1")
df.head()

#df.info() # il y a 195 colonnes et 8 378 lignes
#df.dtypes # pour voir les types des colonnes

,iid,id,gender,idg,condtn,wave,round,position,positin1,order,...,attr3_3,sinc3_3,intel3_3,fun3_3,amb3_3,attr5_3,sinc5_3,intel5_3,fun5_3,amb5_3
0,1,1.0,0,1,1,1,10,7,NaN,4,...,5.0,7.0,7.0,7.0,7.0,NaN,NaN,NaN,NaN,NaN
1,1,1.0,0,1,1,1,10,7,NaN,3,...,5.0,7.0,7.0,7.0,7.0,NaN,NaN,NaN,NaN,NaN
2,1,1.0,0,1,1,1,10,7,NaN,10,...,5.0,7.0,7.0,7.0,7.0,NaN,NaN,NaN,NaN,NaN
3,1,1.0,0,1,1,1,10,7,NaN,5,...,5.0,7.0,7.0,7.0,7.0,NaN,NaN,NaN,NaN,NaN
4,1,1.0,0,1,1,1,10,7,NaN,7,...,5.0,7.0,7.0,7.0,7.0,NaN,NaN,NaN,NaN,NaN


In [31]:
# 2. STATISTIQUES DESCRIPTIVES DU DATASET 
# ---------------------------------------------------------------- 

df.describe(include="all")
#df["order"].describe(include="all")
#df.isna().sum() 
# attention: il y a beaucoups de valeurs manquantes dans certaines colonnes, 
# il faudra les traiter avant de faire des analyses.


,iid,id,gender,idg,condtn,wave,round,position,positin1,order,...,attr3_3,sinc3_3,intel3_3,fun3_3,amb3_3,attr5_3,sinc5_3,intel5_3,fun5_3,amb5_3
count,8378.000000,8377.000000,8378.000000,8378.000000,8378.000000,8378.000000,8378.000000,8378.000000,6532.000000,8378.000000,...,3974.000000,3974.000000,3974.000000,3974.000000,3974.000000,2016.000000,2016.000000,2016.000000,2016.000000,2016.000000
unique,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
top,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
freq,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
mean,283.675937,8.960248,0.500597,17.327166,1.828837,11.350919,16.872046,9.042731,9.295775,8.927668,...,7.240312,8.093357,8.388777,7.658782,7.391545,6.810020,7.615079,7.932540,7.155258,7.048611
std,158.583367,5.491329,0.500029,10.940735,0.376673,5.995903,4.358458,5.514939,5.650199,5.477009,...,1.576596,1.610309,1.459094,1.744670,1.961417,1.507341,1.504551,1.340868,1.672787,1.717988
min,1.000000,1.000000,0.000000,1.000000,1.000000,1.000000,5.000000,1.000000,1.000000,1.000000,...,2.000000,2.000000,3.000000,2.000000,1.000000,2.000000,2.000000,4.000000,1.000000,1.000000
25%,154.000000,4.000000,0.000000,8.000000,2.000000,7.000000,14.000000,4.000000,4.000000,4.000000,...,7.000000,7.000000,8.000000,7.000000,6.000000,6.000000,7.000000,7.000000,6.000000,6.000000
50%,281.000000,8.000000,1.000000,16.000000,2.000000,11.000000,18.000000,8.000000,9.000000,8.000000,...,7.000000,8.000000,8.000000,8.000000,8.000000,7.000000,8.000000,8.000000,7.000000,7.000000
75%,407.000000,13.000000,1.000000,26.000000,2.000000,15.000000,20.000000,13.000000,14.000000,13.000000,...,8.000000,9.000000,9.000000,9.000000,9.000000,8.000000,9.000000,9.000000,8.000000,8.000000


In [48]:
# 3. NETTOYAGE DES DONNEES
# ---------------------------------------------------------------- 

# Nettoyage des données manquantes et abérantes pour les variables d'intérets:
# match, dec_o, gender, age, samerace
# attr1_1, sinc1_1, intel1_1, fun1_1, amb1_1, shar1_1
# attr2_1, sinc2_1, intel2_1, fun2_1, amb2_1, shar2_1
# attr4_1, sinc4_1, intel4_1, fun4_1, amb4_1, shar4_1
# attr3_1, sinc3_1, intel3_1, fun3_1, amb3_1, shar3_1
# attr5_1, sinc5_1, intel5_1, fun5_1, amb5_1, shar5_1
# attr_o, sinc_o, intel_o, fun_o, amb_o, shar_o
# attr, sinc, intel, fun, amb, shar

# Normalisation des vagues 6 à 9
# Les vagues 6-9 utilisent une échelle 1-10 → on les convertit en 1-100

vagues_10 = [6, 7, 8, 9]
cols_to_fix = ['attr1_1', 'sinc1_1', 'intel1_1', 'fun1_1', 'amb1_1', 'shar1_1']

# On identifie uniquement les lignes concernées
mask = df['wave'].isin(vagues_10)

# On vérifie si les valeurs sont bien sur 1-10 (sécurité)
if df.loc[mask, cols_to_fix].max().max() <= 10:
    df.loc[mask, cols_to_fix] = df.loc[mask, cols_to_fix] * 10
    print("Normalisation effectuée pour les vagues 6 à 9.")
else:
    print("Les vagues 6 à 9 semblent déjà normalisées.")


# Nettoyage des revenus (String to Float)
if df['income'].dtype == 'object':
    df['income'] = df['income'].str.replace(',', '').astype(float)
df['income'] = df['income'].fillna(df['income'].median())


Les vagues 6 à 9 semblent déjà normalisées.


In [33]:
# 4. Créer le genre pour le partner
df['gender_o'] = 1 - df['gender']

df["gender"] = df["gender"].replace({0: "Female", 1: "Male"})
df["gender_o"] = df["gender_o"].replace({0: "Female", 1: "Male"})
df.head()

,iid,id,gender,idg,condtn,wave,round,position,positin1,order,...,sinc3_3,intel3_3,fun3_3,amb3_3,attr5_3,sinc5_3,intel5_3,fun5_3,amb5_3,gender_o
0,1,1.0,Female,1,1,1,10,7,NaN,4,...,7.0,7.0,7.0,7.0,NaN,NaN,NaN,NaN,NaN,Male
1,1,1.0,Female,1,1,1,10,7,NaN,3,...,7.0,7.0,7.0,7.0,NaN,NaN,NaN,NaN,NaN,Male
2,1,1.0,Female,1,1,1,10,7,NaN,10,...,7.0,7.0,7.0,7.0,NaN,NaN,NaN,NaN,NaN,Male
3,1,1.0,Female,1,1,1,10,7,NaN,5,...,7.0,7.0,7.0,7.0,NaN,NaN,NaN,NaN,NaN,Male
4,1,1.0,Female,1,1,1,10,7,NaN,7,...,7.0,7.0,7.0,7.0,NaN,NaN,NaN,NaN,NaN,Male


In [34]:
# 5. Création du dataset par individu (pour éviter les doublons)
profiles = df.drop_duplicates(subset='iid')

In [35]:
# 6. Description de l'age par genre
profiles.groupby('gender')['age'].describe()

# boxplot de l'age par genre
fig=px.box(
    profiles, 
    x="gender", 
    y="age", 
    color="gender", 
    title="Distribution de l'âge par genre")
fig.show()

# histogramme de l'age par genre 
fig = px.histogram(
    profiles, 
    x="age", 
    color="gender",
    title="Distribution de l'âge par genre",
    labels={"age": "Âge", "gender_label": "Genre"},
    color_discrete_map={"Female": "red", "Male": "blue"},
    #opacity=0.7
)
fig.show()

Les femmes ont entre 19 et 55 ans avec une médiane de 26 ans et les hommes ont entre 18 et 42 ans avec une médiane de 27 ans.

In [36]:
# 7. DESCRIPTION DES AUTRES VARIABLES ET SPEED DATING
# ---------------------------------------------------------------- 

# wave: entre 1 et 21
# round: number of people that met in wave -> entre 5 et 22
# position: station number where met partner -> entre 1 et 22
# order: the number of date that night when met partner -> entre 1 et 22

# VARIABLE CIBLE: 
# "match"
# 1 = la personne veut revoir l’autre,
# 0 = non

# VARIABLE DECISION POUR REVOIR L'AUTRE: 
# "dec_o"

# six attributes: Attractiveness, Sincerity, Intelligence, Fun, Ambition, and Shared Interests.

## TIME 1
# what do you want?
# attr1_1, sinc1_1, intel1_1, fun1_1, amb1_1, shar1_1

# What do you think the opposite sex looks for in a date?
# attr2_1, sinc2_1, intel2_1, fun2_1, amb2_1, shar2_1

# what you think MOST of your fellow men/women look for…?
# attr4_1, sinc4_1, intel4_1, fun4_1, amb4_1, shar4_1

# Self-assessment 
# attr3_1, sinc3_1, intel3_1, fun3_1, amb3_1, shar3_1

# How I think others see me
# attr5_1, sinc5_1, intel5_1, fun5_1, amb5_1, shar5_1

# partner
# attr_o, sinc_o, intel_o, fun_o, amb_o, shar_o

In [37]:
# Description du taux de match
fig_f = px.pie(
    df["match"].value_counts(),
    names=df["match"].value_counts().index.map({0: "Pas de match", 1: "Match"}),
    values=df["match"].value_counts().values,
    title="Répartition des matchs")
fig_f.show()

# Calcul du taux de decision pour un second date par genre
dec_by_gender = df.groupby("gender")["dec"].value_counts().unstack().fillna(0)

# Camembert pour les femmes
fig_f = px.pie(
    dec_by_gender.loc["Female"],
    names=dec_by_gender.loc["Female"].index.map({0: "Pas de match", 1: "Match"}),
    values=dec_by_gender.loc["Female"].values,
    title="Répartition des décisions pour un second date chez les femmes"
)
fig_f.show()

# Camembert pour les hommes
fig_h = px.pie(
    dec_by_gender.loc["Male"],
    names=dec_by_gender.loc["Male"].index.map({0: "Pas de match", 1: "Match"}),
    values=dec_by_gender.loc["Male"].values,
    title="Répartition des décisions pour un second date chez les hommes"
)
fig_h.show()


Il y a donc 16.5% de match. Quand à la décision pour un second date, il y'a 36.5% de oui chez les femmes contre 47.4% chez les hommes.

# PART 2: HELPERS

* What are the least desirable attributes in a male partner? Does this differ for female partners?
* How important do people think attractiveness is in potential mate selection vs. its real impact?
* Are shared interests more important than a shared racial background?
* Can people accurately predict their own perceived value in the dating market?
* In terms of getting a second date, is it better to be someone's first speed date of the night or their last?

### Q1: What are the least desirable attributes in a male partner? Does this differ for female partners?

In [38]:
# Calculer la moyenne des attributs pour les partenaire hommes
attributes_o = ["attr_o", "sinc_o", "intel_o", "fun_o", "amb_o", "shar_o"]

mean_male_o = profiles[profiles["gender_o"] == "Male"][attributes_o].mean().reset_index()
mean_male_o.columns = ["attribute", "mean_male_score"]
mean_male_o = mean_male_o.sort_values(by="mean_male_score", ascending=True)
print(mean_male_o)
print("The least desirable attribute for a male partner is Shared Interest")
print("-------------------------------")

# Calculer la moyenne des attributs pour les partenaire femmes
mean_female_o = profiles[profiles["gender_o"] == "Female"][attributes_o].mean().reset_index()
mean_female_o.columns = ["attribute", "mean_female_score"]
mean_female_o = mean_female_o.sort_values(by="mean_female_score", ascending=True)
print(mean_female_o)
print("The least desirable attribute for a female partner is Shared Interest")

  attribute  mean_male_score
5    shar_o         5.790875
0    attr_o         7.112132
3     fun_o         7.133829
4     amb_o         7.326996
2   intel_o         7.979630
1    sinc_o         8.103321
The least desirable attribute for a male partner is Shared Interest
-------------------------------
  attribute  mean_female_score
5    shar_o           5.155378
0    attr_o           5.461255
3     fun_o           6.202247
1    sinc_o           6.957364
4     amb_o           7.156000
2   intel_o           7.382463
The least desirable attribute for a female partner is Shared Interest


In [39]:
# Utilisation de subplots pour comparer Hommes (1) et Femmes (0)
fig = make_subplots(rows=1, cols=2, subplot_titles=("Préférences : Hommes", "Préférences : Femmes"))

pref_cols = ['attr1_1', 'sinc1_1', 'intel1_1', 'fun1_1', 'amb1_1', 'shar1_1']
names = ['Attractivité', 'Sincérité', 'Intelligence', 'Humour', 'Ambition', 'Intérêts']

fig.add_trace(go.Bar(x=names, y=profiles[profiles.gender=="Male"][pref_cols].mean(), name="H"), row=1, col=1)
fig.add_trace(go.Bar(x=names, y=profiles[profiles.gender=="Female"][pref_cols].mean(), name="F"), row=1, col=2)

fig.update_layout(title_text="Attributs les moins désirables (Déclaration d'intention)", showlegend=False)
fig.show()


In [40]:
# HEATMAP
# 1. On groupe par genre et on calcule la moyenne des attributs
heatmap_data = profiles.groupby('gender')[pref_cols].mean()

# 2. On renomme pour que ce soit lisible (Standard Plotly)
heatmap_data.columns = ['Beauté', 'Sincérité', 'Intelligence', 'Humour', 'Ambition', 'Intérêts']
heatmap_data.index = ['Femmes', 'Hommes']

# 3. Création de la Heatmap (Plus précise qu'une corrélation ici)
import plotly.express as px
fig = px.imshow(
    heatmap_data,
    text_auto=True, 
    aspect="auto",
    color_continuous_scale='RdPu',
    title="<b>Profil type des attentes :</b> Score moyen attribué par genre",
    labels=dict(x="Attributs", y="Genre", color="Importance (0-100)")
)
fig.show()

Conclusion Q1: L'analyse montre une différence entre les genres : alors que les hommes placent l'attractivité loin devant tous les autres critères, les femmes expriment des attentes plus équilibrées entre la sincérité, l'intelligence et l'humour. Pour les deux genres, l'ambition et les intérêts communs sont déclarés comme les moins prioritaires.

### Q2: How important do people think attractiveness is in potential mate selection vs. its real impact?

In [41]:
# variables to analyze : attr1_1 vs attr_o
# variable of interest : match

#Q2_1
df_plot = df.copy()
df_plot["match_label"] = df_plot["match"].map({0: "Non", 1: "Oui"})

fig = px.box(
    df_plot,
    x="match_label",
    y="attr1_1",
    color="match_label",
    title="Importance déclarée de l’attractivité selon match"
)
fig.show()

fig2 = px.box(
    df_plot,
    x="match_label",
    y="attr_o",
    color="match_label",
    title="Attractivité réelle reçue selon match"
)
fig2.show()

#Q2_3
df[["attr1_1", "attr_o", "match"]].corr()
df[['attr_o', 'match']].corr()

,attr_o,match
attr_o,1.000000,0.260837
match,0.260837,1.000000


In [42]:
# Visualisation de la "Probabilité de succès"
fig_scatter = px.scatter(
    df, 
    x='attr_o', 
    y='prob_o', 
    color='attr_o', 
    color_continuous_scale='Blues',
    trendline="ols",
    title="<b>Lien de causalité :</b> Plus on est jugé beau, plus on pense avoir de chances",
    labels={'attr_o': 'Note de Beauté reçue', 'prob_o': 'Chance de succès estimée'},
    opacity=0.7
)
fig_scatter.show()

Conclusion Q2:
Les graphiques montrent que l’attractivité joue un rôle important dans le fait de matcher.
Les participants qui obtiennent un match sont en moyenne jugés plus attirants par leurs partenaires, et ils accordent aussi légèrement plus d’importance à l’apparence.
De plus, plus une personne est perçue comme belle, plus elle estime avoir des chances de réussir, ce qui peut influencer positivement son comportement.

En résumé : l’attractivité est clairement un facteur déterminant dans la probabilité de match.


### Q3: Are shared interests more important than a shared racial background?

In [24]:
# for shared interest, variable: shar_o or shar1_1
# for same race, variable: samerace

In [44]:
# Q3_1 — Taux de match selon même origine

# Calcul du taux de match
race_match = (
    df.groupby("samerace")["match"]
      .mean()
      .reset_index()
)

# Renommage pour plus de clarté
race_match["samerace"] = race_match["samerace"].map({
    0: "Origine différente",
    1: "Même origine"
})

print("Taux de match selon l'origine :")
print(race_match)

# Visualisation simple
fig = px.bar(
    race_match,
    x="samerace",
    y="match",
    title="Taux de match selon l'origine",
    text="match"
)

# Affichage en pourcentage
fig.update_traces(texttemplate="%{text:.1%}", textposition="outside")
fig.update_yaxes(title="Taux de match", tickformat=".0%")
fig.update_xaxes(title="Origine")

fig.show()


Taux de match selon l'origine :
             samerace     match
0  Origine différente  0.160806
1        Même origine  0.170688


Le taux de match est presque identique entre :
- Origine différente : 16.1%
- Même origine : 17.1%

La différence est d’environ 1 point, ce qui indique que la race n’a quasiment aucun effet sur la probabilité de match.

In [45]:
# label pour match
df["match_label"] = df["match"].map({0: "No match", 1: "Match"})

# Q3_2 Share interest
match_shar=df.groupby("match_label")[["shar1_1", "shar_o"]].mean()
print(match_shar)

# Q3_3
fig=px.box(df_plot, x="match_label", y="shar1_1", color="match_label", title="Shared interest importance by match")
fig.show()

fig2=px.box(df_plot, x="match_label", y="shar_o", color="match_label", title="Shared interest importance for the partner by match")
fig2.show()

#Q3_4
df[["shar1_1", "shar_o", "match"]].corr()

               shar1_1    shar_o
match_label                     
Match        11.152882  6.753571
No match     11.981609  5.208209


,shar1_1,shar_o,match
shar1_1,1.000000,-0.03385,-0.048318
shar_o,-0.033850,1.00000,0.270840
match,-0.048318,0.27084,1.000000


Les boxplots montrent que :
- les personnes qui matchent déclarent accorder plus d’importance aux intérêts communs (shar1_1)
- elles reçoivent aussi des scores plus élevés de la part de leurs partenaires (shar_o)

La corrélation confirme :
- shar_o ↔ match = 0.27 → effet modéré et réel
- shar1_1 ↔ match ≈ 0 → effet très faible

In [46]:
# RACE VS INTÉRÊTS

# Comparons le taux de match si même race vs corrélation d'intérêts
race_impact = df.groupby('samerace')['match'].mean()

# Utilisation d'un Box plot pour voir la distribution des intérêts chez ceux qui matchent
fig_race = px.box(df, x='match', y='int_corr', color='samerace',
             title="Impact de la similarité raciale vs Intérêts communs",
             labels={'int_corr': 'Corrélation des intérêts', 'match': 'Match (0/1)'})
fig_race.show()

Conclusion Q3: Les résultats montrent que l’origine (même race ou non) a un effet très faible sur le taux de match : la différence entre les deux groupes est d’environ 1 point de pourcentage, ce qui est négligeable.
En revanche, les deux boxplots indiquent que les personnes qui ont matché ont tendance à accorder plus d’importance aux intérêts partagés, surtout lorsqu’il s’agit de l’importance qu’elles attribuent aux intérêts du partenaire.
Les intérêts partagés semblent jouer un rôle plus important que l’origine dans la probabilité de match.

### Q4: Can people accurately predict their own perceived value in the dating market?

In [29]:
# six attributes: Attractiveness, Sincerity, Intelligence, Fun, Ambition, and Shared Interests.
# attr, sinc, intel, fun, amb, shar

# Ce que les gens pensent d’eux-mêmes
# attr1_1 : importance qu’ils accordent à l’attractivité
# attr : comment ils s’auto‑évaluent (selon les versions du dataset)
# Ce que les autres pensent d’eux
# attr_o : attractivité perçue par le partenaire
# match : résultat réel

In [49]:
#Q4_1

print("Attractiveness: self vs partner rating by match:")
print(df.groupby("match")[["attr", "attr1_1", "attr_o"]].mean(), "\n")
print("------------------------------")
print("Sincerity: self vs partner rating by match:")
print(df.groupby("match")[["sinc", "sinc1_1", "sinc_o"]].mean(), "\n")
print("------------------------------")
print("Intelligence: self vs partner rating by match:")
print(df.groupby("match")[["intel", "intel1_1", "intel_o"]].mean(), "\n")
print("------------------------------")
print("Fun: self vs partner rating by match:")
print(df.groupby("match")[["fun", "fun1_1", "fun_o"]].mean(), "\n")
print("------------------------------")
print("Ambition: self vs partner rating by match:")
print(df.groupby("match")[["amb", "amb1_1", "amb_o"]].mean(), "\n")
print("------------------------------")
print("Shared interests: self vs partner rating by match:")
print(df.groupby("match")[["shar", "shar1_1", "shar_o"]].mean(), "\n")

Attractiveness: self vs partner rating by match:
           attr    attr1_1    attr_o
match                               
0      5.961230  22.431144  5.961320
1      7.319535  22.938740  7.319898 

------------------------------
Sincerity: self vs partner rating by match:
           sinc    sinc1_1    sinc_o
match                               
0      7.046056  17.497447  7.045975
1      7.809489  16.883026  7.809489 

------------------------------
Intelligence: self vs partner rating by match:
          intel   intel1_1   intel_o
match                               
0      7.249851  20.223219  7.250522
1      7.950365  20.480967  7.950365 

------------------------------
Fun: self vs partner rating by match:
            fun     fun1_1     fun_o
match                               
0      6.155185  17.344609  6.154817
1      7.598535  18.028371  7.598535 

------------------------------
Ambition: self vs partner rating by match:
            amb     amb1_1     amb_o
match             

- les personnes qui matchent ont des self‑ratings plus élevés
- les personnes qui ne matchent pas ont des self‑ratings plus bas   

Cela montre que les gens qui plaisent réellement ont tendance à avoir une meilleure perception d’eux‑mêmes.

In [ ]:
# Q4_2 

fig = px.scatter(
    df,
    x="attr1_1",
    y="attr_o",
    trendline="ols",
    title="Self-rated vs partner-rated attractiveness"
)
fig.show()

Le scatter plot montre :
- un nuage de points aligné positivement
- une droite de régression ascendante
- une relation faible à modérée mais réelle
👉 Les personnes qui se trouvent attirantes sont aussi jugées attirantes par les partenaires.

Conclusion Q4: Les participants semblent relativement précis dans l’évaluation de leur propre valeur sur le marché du dating.
Les self‑ratings sont fortement alignés avec les ratings donnés par les partenaires, comme le montrent les moyennes et le scatterplot.
Cependant, ils ont tendance à surestimer l’importance de leurs qualités (importance déclarée très élevée), ce qui montre une légère distorsion entre ce qu’ils pensent être déterminant et ce qui influence réellement le match.
👉 En résumé : les gens évaluent assez bien comment ils sont perçus, mais surestiment l’importance de leurs propres attributs dans le succès du dating.


### Q5: In terms of getting a second date, is it better to be someone's? First speed date of the night or their last?

In [ ]:
# Si les premières positions ont un taux plus élevé → avantage au début
# Si les dernières positions ont un taux plus élevé → avantage à la fin
# Si c’est plat → aucun effet

print("Match rate for each position number where met partner:")
df.groupby("position")["match"].mean()

Match rate for each position number where met partner:


position
1     0.163498
2     0.155235
3     0.176895
4     0.121324
5     0.180392
6     0.172794
7     0.191235
8     0.154440
9     0.159363
10    0.219298
11    0.243523
12    0.160622
13    0.108808
14    0.100559
15    0.230303
16    0.153333
17    0.149254
18    0.149254
19    0.144330
20    0.086420
21    0.139535
22    0.272727
Name: match, dtype: float64

👉 Il n’y a aucune tendance croissante ou décroissante.
Les variations semblent aléatoires, probablement dues à la petite taille des groupes.

In [50]:
# Q5_1
fig = px.box(
    df,
    x="position",
    y="attr_o",
    title="Attractivité perçue par le partenaire selon l'ordre du rendez-vous",
    labels={"position": "Ordre dans la soirée", "attr_o": "Attractivité perçue par le partenaire"}
)

fig.show()

Les boxplots montrent que :
- l’attractivité perçue ne change pas selon l’ordre du rendez‑vous
- les médianes sont très proches
- aucune position ne ressort comme plus avantageuse

👉 Les participants ne deviennent ni plus généreux ni plus sévères au fil de la soirée


In [51]:
# correlation entre position et match
df[["position", "match"]].corr()


,position,match
position,1.00000,-0.01021
match,-0.01021,1.00000


👉 Aucun lien statistique entre l’ordre du rendez‑vous et la probabilité de match


In [ ]:
# EFFET DE L'ORDRE 
# ----------------------------------------------------------------

# Q5_2

order_effect = df.groupby('order')['dec'].mean().reset_index()
fig_order = px.line(order_effect, x='order', y='dec', 
               title="Évolution du taux de 'OUI' au fil de la soirée",
               markers=True)
fig_order.show()

Le graphique “Évolution du taux de OUI” montre :

👉 Les participants ne disent pas plus “oui” au début ou à la fin

Conclusion Q5: L’ordre du speed‑date n’a pas d’impact réel sur les chances d’obtenir un second rendez‑vous.
L’attractivité perçue par le partenaire reste stable quel que soit l’ordre, et la corrélation entre la position et le match est quasi nulle (r = –0.01).
Même si les taux de match varient légèrement d’une position à l’autre, ces fluctuations ne suivent aucune tendance claire.
Être le premier, le dernier ou au milieu de la soirée ne donne donc aucun avantage particulier pour obtenir un second date.


# PART 3: FINAL QUESTION
## what makes people interested into each other to go on a second date


⭐ Sur l’ensemble de l’expérience, le taux de match est de 16,5 %, soit environ 1 speed‑date sur 6 qui aboutit à un accord mutuel pour un second rendez‑vous.
Les femmes disent “oui” dans 36,5 % des cas, contre 47,4 % pour les hommes.

⭐ Contrairement aux idées reçues, ce ne sont ni la race, ni l’ordre du rendez‑vous, ni l’ambition qui déterminent le match.
Les facteurs qui ressortent clairement sont :

✔️ Shared interests perçus par le partenaire.
C’est le facteur le plus prédictif du match.

✔️ Fun et attractivité perçue.
Ils jouent un rôle réel mais modéré.
Les personnes jugées plus fun ou plus attirantes ont légèrement plus de chances de matcher.

✔️ Sincérité
Moins forte que fun et shared interests, mais présente dans les décisions féminines.
Les participants évaluent assez bien comment ils sont perçus.
Ils prédisent correctement leur valeur perçue.
Mais ils surestiment l’importance de leurs qualités (*_1_1 très élevés).
Cela peut créer un décalage entre attentes et réalité.

⭐ Recommandations pour améliorer le taux de match
✔️ Mettre en avant les intérêts communs
Les plateformes pourraient proposer :
- des centres d’intérêt plus visibles
- des activités partagées
- des ice‑breakers basés sur les hobbies
✔️ Valoriser le fun et la spontanéité
Les profils plus “fun” matchent davantage →
inciter à montrer humour, énergie, personnalité.
✔️ Aider les utilisateurs à calibrer leurs attentes
Puisque les gens surestiment l’importance de leurs qualités :
- proposer des feedbacks anonymisés
- donner des conseils de présentation réalistes
- encourager une perception de soi plus alignée avec la réalité
✔️ Ne pas se focaliser sur des facteurs inutiles
Race, ordre du rendez‑vous, ambition déclarée → aucun impact réel sur le match.

⭐ Conclusion finale
Les résultats montrent que ce qui crée l’envie d’un second rendez‑vous n’est pas la beauté seule, ni l’origine, ni le timing, mais la compatibilité perçue : shared interests, fun, sincérité, et une perception mutuelle positive.

👉 Les gens se plaisent quand ils se sentent compris, alignés et à l’aise — pas seulement quand ils se trouvent beaux.
